### DISCUSSION TITLE
Npc not attacking the player that activated the event

### DISCUSSION THREAD

**Discussion by Gibier1** [AscensionGameDev/Intersect-Engine#2262] [Ideas]
It would be great to add the options in the event editor 'spawn npc' a case 'Is friend' (with party option too)

This way the engine's user could make a working invocator class.

**Comment by WeylonSantana** [comment]
This is not the most correct way to try to create a summoner class, an NPC that does not attack its "summoner" leaves a lot to be desired. 

1 - Movement
You should follow your "summoner", you should surround your "summoner" when he is standing still, you should overcome the intersect barriers when it comes to his floor limit settings

2 - Combat
It should attack the NPCs that target the player, it should have a toggleable option as to whether this summoned NPC could be attacked or not, it should know when to apply buffs and cures to the player 

3 - More
It's normal in some games for people to be around you collecting loot and the like, so there could be similar things for summoners. Also have a summoning limit, and so on.

So this suggestion is something that I don't see coming into the base of the engine, to try to make a detour to try to have support for the summoner class

But we should actually have support for this type of class, if the engine currently offers to be an enine for creating rpgs, then summoners have always been part of rpgs.

But the implementation for this would also be basically the same implementation for a pet system
